dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py

```diff
@@ -1205,6 +1202,130 @@ def read_variables_txt_file(self):
+                not self.is_der_residual_vars(name_var_eval) and not self.is_assign_residual_vars(name_var_eval):
+                continue
+
+            list_depend = [] # list of vars on which depends the function
+            is_elligible = True
+            for line in f.get_body():
+                if "STATE_DER" in line:
+                    is_elligible = False
+                for func_name in name_func_to_search:
+                    if func_name +"("  in line or func_name +" (" in line:
+                        is_elligible = False
+                        func = name_func_to_search[func_name]
+                        outputs = func.find_outputs_from_call(line)
+                        if(len(outputs) >= 1):
+                            name_var_eval = outputs[0]
+                        inputs = func.find_inputs_from_call(line)
+                        list_depend.extend(inputs)
+            if name_var_eval is not None:
+                list_depend.append(name_var_eval) # The / equation function depends on the var it evaluates
+                if name_var_eval in map_dep.keys():
+                    list_depend.extend( map_dep[name_var_eval] ) # We get the other vars (from *._info.xml)
+                if is_elligible:
+                    function_to_eval_variable[f] = name_var_eval
+                for var_name in list_depend:
+                    if var_name == "time": continue
+                    if self.is_residual_vars(var_name) : continue
+                    var_list = filter(lambda x: (x.get_name() == var_name),self.list_vars)
+                    assert(len(var_list) <= 1)
+                    var = None
+                    if len(var_list) == 1:
+                        var = var_list[0]
+                    if (var is None): continue
+                    if is_discrete_real_var(var): continue
+                    if is_integer_var(var): continue
+                    if is_bool_var(var): continue
+                    if "$whenCondition" in var_name : continue
+                    if not var_name in variable_to_equation_dependencies:
+                        variable_to_equation_dependencies[var_name] = []
+                    variable_to_equation_dependencies[var_name].append(f_num_omc)
+
+        function_to_remove = []
+        for f in function_to_eval_variable:
+            var_name = function_to_eval_variable[f]
+            if var_name in variable_to_equation_dependencies and len( variable_to_equation_dependencies[var_name]) == 1:
+                var_list = filter(lambda x: (x.get_name() == var_name),self.list_vars)
+                assert(len(var_list) <= 1)
+                var = None
+                if len(var_list) == 1:
+                    var = var_list[0]
+                assert(var != None)
+                self.list_complex_calculated_vars[var] = f
+                function_to_remove.append(f)
+                map_var_name_2_addresses[var_name]= "SHOULD NOT BE USED"
```

Comment:
I like the effort on the "SHOULD NOT BE USED" but perhaps we can be even more specific :"SHOULD NOT BE USED - CALC VAR" for example?

Reply:
Done